src/main/java/coloringpage/Converter.java

```diff
@@ -2,69 +2,113 @@
-        for(int i =0; i<blurRad*blurRad; i++) {
-            blurKernel[i] = 1.0f/256.0f;
-        }
+    private BufferedImage invertAndBlurImage(BufferedImage image) {
+        BufferedImage newImage = cloneImage(image);
+        invertImage(newImage);
+        return blurImage(newImage);
+    }
 
+    private void invertImage(BufferedImage image) {
+        int width = image.getWidth();
+        int height = image.getHeight();
 
-        BufferedImage blurryImg = processImage(image);
+        for(int i=0; i<height; i++) {
+            for(int j=0; j<width; j++) {
+                Color c = new Color(image.getRGB(j, i));
+                int red = 255 - c.getRed();
+                int green = 255 - c.getGreen();
+                int blue = 255 - c.getBlue();
 
-        return blurryImg;
+                Color newColor = new Color(red, green, blue);
+                image.setRGB(j, i, newColor.getRGB());
+            }
+        }
     }
-    public  BufferedImage processImage(BufferedImage image) {
+
+    private BufferedImage blurImage(BufferedImage image) {
         /*
         Resources:
         https://www.javatips.net/api/java.awt.image.convolveop
         https://stackoverflow.com/questions/29295929/java-blur-image
-        * */
+        */
+
         int radius = 20;
         int size = radius * 2 + 1;
         float weight = 1.0f / (size * size);
         float[] data = new float[size * size];
 
-        for (int i = 0; i < data.length; i++) {
-            data[i] = weight;
-        }
+        Arrays.fill(data, weight);
         BufferedImageOp blurFilter = new ConvolveOp(new Kernel(size, size, data), ConvolveOp.EDGE_ZERO_FILL, null);
         return blurFilter.filter(image, null);
     }
+
+    private BufferedImage dodge(BufferedImage front, BufferedImage back) {
+        //https://www.freecodecamp.org/news/sketchify-turn-any-image-into-a-pencil-sketch-with-10-lines-of-code-cf67fa4f68ce/
+        BufferedImage result = cloneImage(front);
+
+        int width = front.getWidth();
+        int height = front.getHeight();
+
+        for(int i=0; i<height; i++) {
+            for(int j=0; j<width; j++) {
+                int frontValue = new Color(front.getRGB(j, i)).getRed(); //could be any of RGB, red is arbitrary
+                int backValue = new Color(back.getRGB(j, i)).getRed();
+
+                int newValue = frontValue * 255 / (255 - backValue);
+                if (newValue > 255 || backValue == 255) {
+                    newValue = 255;
+                }
+
+                Color newColor = new Color(newValue, newValue, newValue);
+                result.setRGB(j, i, newColor.getRGB());
+            }
+        }
+
+        return result;
+    }
+
+    private BufferedImage cloneImage(BufferedImage image) {
+        //https://bytenota.com/java-cloning-a-bufferedimage-object/
+        ColorModel colorModel = image.getColorModel();
+        WritableRaster raster = image.copyData(null);
```

Comment:
What is a WritableRaster?

Reply:
apparently there is no easy way to make a deep copy of a BufferedImage, so this is a method I found to copy each part of it individually and put them together. I believe the WriteableRaster part has to do with the actual pixels in the image